pymeasure/experiment/parameters.py

```diff
@@ -287,8 +296,8 @@ def value(self):
 
     @value.setter
     def value(self, value):
-        if self._choices is not None and value in self._choices:
-            self._value = value
+        if self._choices is not None and str(value) in self._choices_dict.keys():
```

Comment:
Think the error raise in case `self._choices` is `None` could be clearer; now if choices is not defined, the error is not very clear on that.

Reply:
I added a separate check for `None` resulting in a more specific error message.